### PR TITLE
Update for TopN

### LINKED ISSUE #294: Document the support of TopN for streaming (mv)
Only rank() but not dense_rank().

### Related code PR

https://github.com/risingwavelabs/risingwave/pull/5116

### Which part(s) of the docs might be affected or should be updated? And how?

Select command or a new page about TopN.

## Test cases

statement ok
create table t(x int, y int);

statement ok
create materialized view mv as 
select x, y from (
    select *, ROW_NUMBER() OVER (PARTITION BY x ORDER BY y) as rank from t
)
where rank <= 3;

statement ok
insert into t values 
    (1, 1), (1, 2), (1, 3), (1, 4), (1, 5),
    (2, 1), (2, 2), (2, 3), (2, 4), (2, 5),
    (3, 1), (3, 2), (3, 3), (3, 4), (3, 5);

query II rowsort
select * from mv;
----
1  1
1  2
1  3
2  1
2  2
2  3
3  1
3  2
3  3

### Reference

https://nightlies.apache.org/flink/flink-docs-release-1.15/docs/dev/table/sql/queries/topn/

### PR DESCRIPTION
**Pull request by bernscode**
Update for Top-N support.

## Info
- **Description**: 
Update for Top-N support. Created a new pull request for issue 294.

- **Related code PR**: 
[ Provide the link to the code PR here. For example, https://github.com/risingwavelabs/risingwave/pull/4085. Delete this part if there's no code PR related. ]

- **Related doc issue**: 
Resolves https://github.com/risingwavelabs/risingwave-docs/issues/294

- **Notes**: 
https://github.com/risingwavelabs/risingwave/pull/5116

## Before merging
  - [ ] (For version-specific PR) I have selected the corresponding software version in **Milestone** and linked the related doc issue to this PR in **Development**.
  - [ ] I have acquired the approval from the owner (and optionally the reviewers) of the code PR and at least one tech writer (`bernscode`, `CharlieSYH`, `emile-00`, & `hengm3467`). 
  - [ ] I have checked the doc site preview, and the updated parts look good. <details><summary>How?</summary>image</details>

## Published doc pages
  [ After merging this PR, edit the description to include the links to the updated doc pages here. For example, https://www.risingwave.dev/docs/latest/intro/. ]

### CODE REVIEW

File: docs/sql/sql-queries.md [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+WHERE rank <= N;
+```
+
+NOTE: `rank` cannot be included in `column_list`.
+
+Parameter Specification:
+
+<ul>
+  <li>
+    ROW_NUMBER(): Assigns a unique, sequential number to each row, starting with one, according to the ordering of rows within the partition. RANK() is also supported for streaming but not batch.
```
Comment (hengm3467) [inline docs/sql/sql-queries.md]: @bernscode Suggest changes: RANK() is also... ---> RANK() is supported for materialized views but not tables.

File: docs/sql/sql-queries.md [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+  <li>
+    PARTITION BY col1[, col2...]: Specifies the partition columns. Each partition will have a Top-N result.
+  </li>
+  <li>
+    ORDER BY col1 [asc|desc][, col2 [asc|desc]...]: Specifies the ordering columns.
+  </li>
+  <li>
+    WHERE is required to recognize the query as a Top-N query.
+  </li>
+</ul>
```
Comment (xxchan) [inline docs/sql/sql-queries.md]:
```suggestion
- `ROW_NUMBER()`: Assigns a unique, sequential number to each row, starting with one, according to the ordering of rows within the partition. `RANK()` is also supported for streaming but not batch.
- `PARTITION BY col1[, col2...]`: Specifies the partition columns. Each partition will have a Top-N result.
- `ORDER BY col1 [asc|desc][, col2 [asc|desc]...]`: Specifies the ordering columns.
- `WHERE rank <= N` is required to recognize the query as a Top-N query.


```

File: docs/sql/sql-queries.md [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+
+Syntax of the Top-N statement:
+```sql
+SELECT [column_list] 
+  FROM (
+    SELECT [column_list], 
+      ROW_NUMBER() OVER ([PARTITION BY col1[, col2...]] 
+        ORDER BY col1 [asc|desc][, col2 [asc|desc]...]) AS rank 
+    FROM table_name)
+WHERE rank <= N;
```
Comment (xxchan) [inline docs/sql/sql-queries.md]:
```suggestion
WHERE rank <= N [AND conditions];
```
is also supported. It seems good to also mention it here.

**Review by xxchan — COMMENTED** [review]
Others LGTM

File: docs/sql/sql-queries.md (hunk 2 above)
Comment (bernscode) [reply docs/sql/sql-queries.md]: @xxchan Excellent recommendation! I'll look to implement this styling for bullet points going forward. I've made this update to the new commit. Thanks!

File: docs/sql/sql-queries.md (hunk 3 above)
Comment (bernscode) [reply docs/sql/sql-queries.md]: @xxchan Sounds good! I've made this update to the new commit. Thanks!

**Review by hengm3467 — APPROVED** [review]
Looks good to me.

File: docs/sql/sql-queries.md (hunk 1 above)
Comment (bernscode) [reply docs/sql/sql-queries.md]: @hengm3467 Good idea! I've made the update to the latest commit. Thanks!

**Pull request merged**